src/components/StatedBalanceForm.test.tsx

```diff
@@ -0,0 +1,59 @@
+import Enzyme, {mount, shallow} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import React, {ReactElement} from "react";
+import StatedBalanceForm from "./StatedBalanceForm";
+import {saveValues} from "../core/Memory";
+
+
+Enzyme.configure({adapter: new Adapter()});
+
+describe('StatedBalanceForm', () => {
+    it('renders without crashing given the required props', () => {
+        const save_value = jest.fn();
+        const get_value = jest.fn();
+        const wrapper = shallow(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
+        expect((wrapper)).toMatchSnapshot();
+    })
+})
+
+test("When user inputs, then get_values is called ", () => {
+    const save_value = jest.fn();
+    const get_value = jest.fn(() => []);
+    const StatedbalanceForm = mount(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
+    StatedbalanceForm.find("button").simulate("click");
+    expect(get_value).toBeCalled();
+})
+
+test("When user inputs, then save_value is called", () => {
+    const save_value = jest.fn();
+    const get_value = jest.fn(() => []);
+    const StatedbalanceForm = mount(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
+    StatedbalanceForm.find("button").simulate("click");
+    expect(save_value).toBeCalled();
+})
+
+test("SaveValues save values to Memory's empty storage ", () => {
+    const save_value = jest.fn();
+    const get_value = jest.fn(() => []);
+    const currentDate = new Date();
+    const StatedbalanceForm = mount(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
+    StatedbalanceForm.find("input").simulate("change", {target: {value: 21}});
+    StatedbalanceForm.find("button").simulate("click");
+    const {date, value}: { date: Date, value: number } = save_value.mock.calls[0][0][0];
```

Comment:
tip:
To use the full power of destructurizing u can write it as:
const [[[{date, value}]]] = save_value.mock.calls
however it can be unreadable

Reply:
you can create function extractFirstArgumentOfFirstMockExecution so this can be more readable.

Reply:
I've got a type error after doing it. Why is it important and why shouldn't we keep things as they are right know?

Reply:
It is just a tip - a suggestion.

For me destructurizing pattern can sometimes be more readable as it depicts the shape of structure we all restructuring. 

And of course you will get type error as structure : [[[{date, value}]]] is of type [[[{date: Date, value: number}]]]

Reply:
Ok, thanks